src/bcf/record.rs

```diff
@@ -79,6 +79,30 @@ impl NumericUtils for i32 {
     }
 }
 
+/// A trait to allow for seamless use of bytes or integer identifiers for filters
+pub trait FilterId {
+    fn id_from_header(&self, header: &HeaderView) -> Result<Id>;
+    fn is_pass(&self) -> bool;
+}
+
+impl FilterId for [u8] {
+    fn id_from_header(&self, header: &HeaderView) -> Result<Id> {
+        header.name_to_id(self)
+    }
+    fn is_pass(&self) -> bool {
+        matches!(self, b"PASS" | b".")
+    }
+}
+
+impl FilterId for Id {
+    fn id_from_header(&self, _header: &HeaderView) -> Result<Id> {
+        Ok(*self)
+    }
+    fn is_pass(&self) -> bool {
+        *self == Id(0)
```

Comment:
In the previous code, this was simply checking `*self == 0` (`*flt_id == 0`) without the `Id()` wrapping. Is this definitely needed, or could this be contributing to overhead, as well?